### ISSUE TITLE
Support `@attr` decorator without parens

### ISSUE THREAD

**Issue by chancancode** [emberjs/data#6328]
Before...

```js
import Model, { attr } from '@ember-data/model';

export default class RentalModel extends Model {
  @attr() title;
  @attr() owner;
  @attr() city;
  @attr() location;
  @attr() category;
  @attr() image;
  @attr() bedrooms;
  @attr() description;
}
```

After...

```js
import Model, { attr } from '@ember-data/model';

export default class RentalModel extends Model {
  @attr title;
  @attr owner;
  @attr city;
  @attr location;
  @attr category;
  @attr image;
  @attr bedrooms;
  @attr description;
}
```

The parens version is still supported and required when you need to specify a transform type.

This matches how the built-in Ember decorators work, which makes it easier to explain `@attr` in the guides, building on what they already learned from the Ember side, without having to explain the gratuitous difference.

**Comment by runspired** [comment]
Did we determine whether this was even possible? It wasn't last I looked into it due to our required support matrix.

**Comment by pzuraq** [comment]
It should be possible, using compatibility helpers if necessary.

**Comment by runspired** [comment]
resolved by #6339 for the most recent ember versions (not 3.8 LTS and likely not some versions thereafter) cc @pzuraq
